scipy/special/special/specfun/specfun.h

```diff
@@ -7696,11 +7691,11 @@ inline void rmn2sp(int m, int n, double c, double x, double cv, int kd, double *
     double ip, nm1, nm, nm2, su0, sw, sd0, su1, sd1, sd2, ga, r1, r2, r3,\
            sf, gb, spl, gc, sd, r4, spd1, spd2, su2, ck1, ck2;
 
-    double *pm = (double *) malloc(252*sizeof(double));
-    double *pd = (double *) malloc(252*sizeof(double));
-    double *qm = (double *) malloc(252*sizeof(double));
-    double *qd = (double *) malloc(252*sizeof(double));
-    double *dn = (double *) malloc(201*sizeof(double));
+    double pm[252];
```

Comment:
The others are probably fine but this one is definitely on the blurry line of *maybe this is too big for the stack* limit. If there is a thread that runs a few, say 4, instances of this function we are talking about ~40 kB on the stack which would be better on the heap. For certain limited hardware this is quite large in my opinion. Not that it would start crashing right away.

The even better option is to rewrite this function so that it doesn't use these dummy arrays for every call. But I don't know anything about that unfortunately.

Reply:
Would be great to have some general rule-of-thumb guidance about these stack allocations. Also what kinds of devices we worry about for scipy. 

This is genuine curiosity: am coming from an HPC background, and knowing nothing about wearable computing or whatnot.

Reply:
I really don't know much about limits but typically around ~64kb there is a code smell. It is not about what this function does but about what the rest of the system is doing at that particular moment. It is a general rule of thumb that is regurgitated but maybe I am just a member of a cult without thinking or researching all the way.

Reply:
@ilayn I'm actually happy to do whatever is best practice, but I don't know what that is. I personally don't think those are too big, but if you want me to change them to `std::unique_ptr` I can.

I think the more general solution is to have all these functions accept buffer pointers as parameters, but I don't want to do that until we know how. In that case, I'd still think we should keep a default version of these functions which does allocation.

Reply:
Fine with me, we can always roll back in case something pops up